### ISSUE TITLE
Testing the trained model problem

### ISSUE THREAD

**Issue by dsim174** [mit-han-lab/pvcnn#12]
(edited my question)
Hi. I'm beginner of pytorch.

I'm trying to understand the code.

I finished training the model with kitti dataset without --evaluate option.

But, when I trained with --evaluate option to accurate evaluation, I got a segmentation fault.

Do I get any clues to solve it?

My command was 'python train.py  configs/kitti/frustum/pvcnne.py --devices 0 --evaluate'

Thanks.

**Comment by synxlin** [comment]
Hi, @dsim174. Could you please provide more information on your error? The evaluation command seems correct to me. 

The evaluation function script is [here](https://github.com/mit-han-lab/pvcnn/blob/master/evaluate/kitti/frustum/eval.py). You can insert `import IPython; IPython.embed()` anywhere you want to insert the breakpoint so that you can check whether different parts of the code have run or not.

**Comment by dsim174** [comment]
Thanks for your reply. I would try to debug in that way.

But, during the debug, the other question arises.

I found that you sampled the points randomly (1k) from the point cloud (~1.5k) in frustum kitti datasets as the input data. Is it right? Then, didn't you use the farthest point sampling (Qi, PointNet)? I think that this random sampling may cause some accuracy issues on the large point cloud datasets.

Thanks.

**Comment by synxlin** [comment]
Hi, @dsim174.

1. Yes, we randomly sample points from the point cloud as model input. This follows Qi's work: Line 171 [here](https://github.com/charlesq34/frustum-pointnets/blob/master/train/provider.py) in his repo of Frustum PointNet implementation based on Tensorflow.

2. We do use the farthest point sampling in Frustum PointNet++. See Line 16 [here](https://github.com/mit-han-lab/pvcnn/blob/master/models/kitti/frustum/segmentation/pointnetpp.py). Note: in Qi's work, farthest point sampling is performed on the model input point cloud which is sampled from the original sensor-collected entire point cloud.
